src/matrix/matrixtdist.jl

```diff
@@ -0,0 +1,194 @@
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::AbstractPDMat)          = MatrixTDist(ν, M, PDMat(Σ), Ω)
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::LinearAlgebra.Cholesky, Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, PDMat(Σ), PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::AbstractMatrix)         = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+MatrixTDist(ν::Real, M::AbstractMatrix, Σ::AbstractPDMat,          Ω::LinearAlgebra.Cholesky) = MatrixTDist(ν, M, Σ,        PDMat(Ω))
+
+#  -----------------------------------------------------------------------------
+#  REPL display
+#  -----------------------------------------------------------------------------
+
+ show(io::IO, d::MatrixTDist) = show_multline(io, d, [(:ν, d.ν), (:M, d.M), (:Σ, Matrix(d.Σ)), (:Ω, Matrix(d.Ω))])
+
+#  -----------------------------------------------------------------------------
+#  Conversion
+#  -----------------------------------------------------------------------------
+
+function convert(::Type{MatrixTDist{T}}, d::MatrixTDist) where T <: Real
+    MM = convert(AbstractArray{T}, d.M)
+    ΣΣ = convert(AbstractArray{T}, d.Σ)
+    ΩΩ = convert(AbstractArray{T}, d.Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(d.ν), MM, ΣΣ, ΩΩ, T(d.c0))
+end
+
+function convert(::Type{MatrixTDist{T}}, ν, M::AbstractMatrix, Σ::AbstractPDMat, Ω::AbstractPDMat, c0) where T <: Real
+    MM = convert(AbstractArray{T}, M)
+    ΣΣ = convert(AbstractArray{T}, Σ)
+    ΩΩ = convert(AbstractArray{T}, Ω)
+    MatrixTDist{T, typeof(MM), typeof(ΣΣ)}(T(ν), MM, ΣΣ, ΩΩ, T(c0))
+end
+
+#  -----------------------------------------------------------------------------
+#  Properties
+#  -----------------------------------------------------------------------------
+
+size(d::MatrixTDist) = size(d.M)
+
+rank(d::MatrixTDist) = minimum( size(d) )
+
+insupport(d::MatrixTDist, X::Matrix) = isreal(X) && size(X) == size(d)
+
+function mean(d::MatrixTDist)
+
+  n, p = size(d)
+
+  (d.ν + p - n > 1) ? (return d.M) : throw(ArgumentError("mean only defined for df + p - n > 1"))
+
+end
+
+mode(d::MatrixTDist) = d.M
+
+params(d::MatrixTDist) = (d.ν, d.M, d.Σ, d.Ω)
+
+@inline partype(d::MatrixTDist{T}) where {T <: Real} = T
+
+#  -----------------------------------------------------------------------------
+#  Evaluation
+#  -----------------------------------------------------------------------------
+
+function _matrixtdist_c0(Σ::AbstractPDMat, Ω::AbstractPDMat, ν::Real)
+
+    n = dim(Σ)
+    p = dim(Ω)
+
+    term1 = logmvgamma(p, (ν + n + p - 1) / 2)
+    term2 = - (n * p / 2) * logπ
+    term3 = - logmvgamma(p, (ν + p - 1) / 2)
+    term4 = (-n / 2) * logdet(Ω)
+    term5 = (-p / 2) * logdet(Σ)
+
+    term1 + term2 + term3 + term4 + term5
+
+end
+
+function logkernel(d::MatrixTDist, X::AbstractMatrix)
+
+    A  = (X - d.M)
+    At = Matrix(A')
```

Comment:
is there any benefit to calling `Matrix` here?

Reply:
I'd love for someone to teach me how to avoid doing this. But such as it is...

`A'` will return a `LinearAlgebra.Adjoint` instance, and `PDMats.jl` doesn't have a `a \ x` method for `a::PDMat` and `x::LinearAlgebra.Adjoint`, so you have to do this to keep `d.Ω \ At` from throwing an error.

Reply:
This works for me without the Matrix call

Reply:
Oh, cool. Sorry about that. I got an error once upon a time.